### DISCUSSION TITLE
Creating Content

### DISCUSSION THREAD

**Discussion by giscus** [hbstack/site#159] [General]
# Creating Content

Content is the most basic and core part of a site. This article will introduce some basic concepts of content and how to create it.

https://hbstack.dev/en/docs/content/create/

**Comment by FuadEfendi** [comment]
I am not sure if it is bug of the framework or not, `hugo new blog/hello/index.md` command just hangs with no response.
`hugo v0.111.3+extended darwin/arm64 BuildDate=unknown`

**Comment by razonyang** [comment]
It probably not an issue of the framework, it's just a set of modules.

There are some useful flags for debuging, e.g. `hugo new blog/hello/index.md -v --debug > debug.log` will echo the log to `debug.log`.

You can also find more topic on the official forum https://discourse.gohugo.io/search?q=hanging.

**Comment by FuadEfendi** [comment]
`hugo mod clean` - to clean cache

`hugo mod` - it will upgrade modules (I had it outdated)

`hugo new blog/hello/index.md -v --debug > debug.log` - hangs again... unpredictably hangs, sometimes shows "downloading modules..." "cvollected modules..." messages, then hangs again. "debug.log" is empty.

I'll report it to Hugo.

**Comment by razonyang** [comment]
Closing it, see https://github.com/orgs/hbstack/discussions/12#discussioncomment-5709489.

**Comment by FuadEfendi** [comment]
What is the best way to create home page? I tried `layout: landing` but it generates very wide page with zero padding on left and right; I tried to add `{.text-center .mb-5}` to the page but it doesn't work as expected. Thanks,

**Reply by razonyang** [reply]
You can find examples and get inspired from https://github.com/hugomods/site, a monorepo.

Such as images.hugomods.com, docker.hugomods.com and so on.

**Comment by FuadEfendi** [comment]
I tried this:
```
{{< html/div class="text-center ml-5 mr-5" >}}

## title
- item1
- tem2

{{< /html/div >}}

I am not sure...

**Reply by razonyang** [reply]
1. Firstly, there is not ml-* mr-*, since they are renamed to ms-* and me-* respectively.
2. If you want to render markdown inside a shortcode, please replace `{{< >}}` with `{{% %}}`, **but it may not 100% persent work, please read the related shortcode doc for detail.**
3. I'd recommend using Bootstrap container to wrap stuff, to make sides are not too close together, learn more on [Bootstrap docs](https://getbootstrap.com/docs) and built-in [Bootstrap shortcodes](https://bootstrap.hugomods.com/).

```markdown
---
title: Homepage
layout: landing
# full_width: false # uncomment to force not to take full_width globally.
---

Remove `breakpoint` to not take full width, it depends on the full_width (page's and global's) parameter.

{{% bs/container breakpoint=fluid %}}

{{% bs/row cols="1" %}}

{{% bs/col %}}
## title
- item1
- tem2
{{% /bs/col %}}

{{% bs/col %}}
I am not sure...
{{% /bs/col %}}

{{% /bs/row %}}

{{% /bs/container %}}

```
